File: PolishPizza/admin/src/types/menuItem.ts
import type { ItemType } from "./enums";

export interface MenuItemResponse {
    id: string;
    itemId: string;
    type: ItemType;
    name: string;
    description: string;
    basePrice: number;
    isAvailable: boolean;
}

export interface MenuItemRequest {
    itemId: string;
    type: ItemType;
    name: string;
    description: string;
    basePrice: number;
    isAvailable: boolean;
}

File: PolishPizza/menu/src/main/java/pizzeria/menu/menuItem/model/MenuItem.java
package pizzeria.menu.menuItem.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.UUID;

@Entity
@Table(name = "menu_items")
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class MenuItem {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private UUID itemId;

    @Enumerated(EnumType.STRING)
    ItemType type;

    private String name;
    private String description;
    private BigDecimal basePrice;

    @Builder.Default
    private Boolean isAvailable = true;
}
